applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -198,51 +203,63 @@ impl wallet_server::Wallet for WalletGrpcServer {
         Ok(Response::new(GetTransactionInfoResponse { transactions }))
     }
 
-    async fn get_all_completed_transactions(
+    async fn get_completed_transactions(
         &self,
-        _request: Request<GetAllCompletedTransactionsRequest>,
-    ) -> Result<Response<GetTransactionInfoResponse>, Status>
+        _request: Request<GetCompletedTransactionsRequest>,
+    ) -> Result<Response<Self::GetCompletedTransactionsStream>, Status>
     {
         debug!(
             target: LOG_TARGET,
             "Incoming GRPC request for GetAllCompletedTransactions"
         );
         let mut transaction_service = self.get_transaction_service();
-        let transactions;
-        match transaction_service.get_completed_transactions().await {
-            Ok(t) => {
-                transactions = t;
-            },
-            Err(_) => {
-                return Err(Status::not_found("No completed transactions found."));
-            },
-        };
+        let transactions = transaction_service
+            .get_completed_transactions()
+            .await
+            .map_err(|err| Status::not_found(format!("No completed transactions found: {:?}", err)))?;
 
-        let transactions = transactions
-            .into_iter()
-            .map(|tx| TransactionInfo {
-                tx_id: tx.1.tx_id,
-                source_pk: tx.1.source_public_key.to_vec(),
-                dest_pk: tx.1.destination_public_key.to_vec(),
-                status: TransactionStatus::from(tx.1.status) as i32,
-                amount: tx.1.amount.into(),
-                is_cancelled: tx.1.cancelled,
-                direction: TransactionDirection::from(tx.1.direction) as i32,
-                fee: tx.1.fee.into(),
-                timestamp: Some(naive_datetime_to_timestamp(tx.1.timestamp)),
-                excess_sig: tx
-                    .1
-                    .transaction
-                    .first_kernel_excess_sig()
-                    .expect("Complete transaction has no kernels")
-                    .get_signature()
-                    .to_vec(),
-                message: tx.1.message,
-                valid: tx.1.valid,
-            })
-            .collect();
+        let (mut sender, receiver) = mpsc::channel(transactions.len());
+        self.executor.spawn(async move {
+            for txn in transactions {
```

Comment:
nit: 

```suggestion
            for (_, txn) in transactions {
```

Then remove all the `.1`. 

This is completely optional, but generally the numeric accessor syntax for tuples is not easy to reason about and should be avoided (what are `things.0` and `things.2`?). Rather "unpack" the tuple (`let (car, _ , boat) = things`)